tensorflow/python/ops/nn_test.py

```diff
@@ -151,6 +153,18 @@ def testGradient(self, x_shape):
     eps = 2e-8
     self.assertLess(err, eps)
 
+  @test_util.run_in_graph_and_eager_modes
+  def testRagged(self):
+    x_list = [[-500, -501, -502], [], [1729, 1729]]
```

Comment:
Add a few more relevant test cases here (including some "regular" ones).

Reply:
Ok, I added a test case with "ordinary values", and split this into cases with "large values" and "short tensors" (a ragged tensor where lengths 0 and 1 occur).